pkg/cloud/services/eks/cluster.go

```diff
@@ -62,8 +62,14 @@ func (s *Service) reconcileCluster(ctx context.Context) error {
 	} else {
 		tagKey := infrav1.ClusterAWSCloudProviderTagKey(eksClusterName)
 		ownedTag := cluster.Tags[tagKey]
-		if ownedTag == nil {
-			return fmt.Errorf("checking owner of %s is %s: %w", s.scope.KubernetesClusterName(), eksClusterName, err)
+		// Prior to https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3573,
+		// Clusters were tagged using s.scope.Name()
+		// To support upgrading older clusters, check for both tags
+		oldTagKey := infrav1.ClusterAWSCloudProviderTagKey(s.scope.Name())
+		oldOwnedTag := cluster.Tags[oldTagKey]
+
+		if ownedTag == nil && oldOwnedTag == nil {
+			return fmt.Errorf("cluster is not tagged with neither %s nor %s", tagKey, oldTagKey)
```

Comment:
Let's use the error message to explain what _should_ be true, since that may be enough information for the user to address the error.

Let's also mention the EKS cluster resource by name, so that the user can look it up using the AWS CLI or Console.

```suggestion
			return fmt.Errorf("EKS cluster resource %q must have a tag with key %q or %q", eksClusterName, oldTagKey, tagKey)
```